transcom/mymove
# MAIN - B-21412 & B-21413 Prime sim - create international crate

## [B-21412](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21412) & [B-21413](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21413)

INT PR: https://github.com/transcom/mymove/pull/14223
Bug fix INT PR: https://github.com/transcom/mymove/pull/14247 

## Summary

This adds the ability for Prime user to create International Crate/Uncrate (ICRT & ICURT) service items similar to Domestic Crate/Uncrate. ICRT includes 2 boolean checkboxes (standalone and external crate) that ICURT does not have. `Market` field with possible values `OCONUS` and `CONUS` has been added to `MTOServiceItemInternationalCrating.yaml` which is derived from corresponding `address.isOconus`.

Also the service item `ICRTSA` has been removed from the DB and code.

- FF under `enable_alaska`.
- Added International Crate and International Uncrate as options when adding service item in Prime Sim.
- `External Crate` boolean checkbox added to International Crating.
- `Market` field added to `MTOServiceItemInternationalCrating.yaml` with enum values: `CONUS` or  `OCONUS`.
- ICRTSA service item has been removed from DB tables (`service_params, re_intl_accessorial_prices, and re_services).
- `external_crate` column added to `mto_service_items` table.

### How to test

1. In your `.envrc` file -> change to `export FEATURE_FLAG_ENABLE_ALASKA=true` to turn on Alaska FF locally.
2. Close your server -> run `direnv allow` and then restart your server.
3. Create a move and get the the step where the Prime can see the move and is able to create a service item (or use `HHGMoveInSITEndsTomorrow` testharness.
4. In Prime Sim -> find the move.
5. Click `Add Service Item` button for the shipment.
6. Select `International Crating` for "Service item type".
7. Select `International Crating` for "Service item code".
8. You should see `Standalone Crate` and `External Crate` options for "International crating" service item code and not "International Uncrating".
9. Test all the options to verify that you are able to create those service items.
10. Make sure item length/width/height is less than the Crate length/width/height.
11. To check if the service item was created -> find the move as TOO and go to move task order tab -> should see it under requested service items table (details will be empty).


## Screenshots
![Screenshot 2024-11-18 at 11 05 57 AM](https://github.com/user-attachments/assets/5ce3560e-7032-4642-8301-da4a8fb7940f)
![Screenshot 2024-11-18 at 11 06 28 AM](https://github.com/user-attachments/assets/a2b3273c-e538-4ef9-bde8-745c50981077)
![Screenshot 2024-11-18 at 12 37 21 PM](https://github.com/user-attachments/assets/ab6a3524-8fc6-4914-9324-3a2627d2a498)
![Screenshot 2024-11-18 at 12 37 27 PM](https://github.com/user-attachments/assets/fbd58b8a-d30f-44d2-82de-506acc48e15f)
![image](https://github.com/user-attachments/assets/b8cda3f6-852c-416e-99b3-f5234291795c)
